Data Structures/Sorting/C++/InsertionSort.cpp

```diff
@@ -0,0 +1,46 @@
+// ********** insertion sort **********
+#include <iostream>
+using namespace std;
+
+/* Insertion sort function*/
+void insertionSort(int arr[], int n)
+{
+	int i, temp, j;
+	for (i = 1; i < n; i++)
+	{
+		temp = arr[i];
+		j = i - 1;
+		
+		while (j >= 0 && arr[j] > temp)
+		{
+			arr[j + 1] = arr[j];
+			j = j - 1;
+		}
+		arr[j + 1] = temp;
+	}
+}
+
+// Function to print array
+void printArray(int arr[], int n)
+{
+	int i;
+	for (i = 0; i < n; i++)
+		cout << arr[i] << " ";
+}
+
+/* Driver code */
+int main()
+{
+	int arr[] = { 22, 10, 15, 4, 8 };
+	int n = sizeof(arr) / sizeof(arr[0]);
+	cout << "The array: ";
+	printArray(arr, n);
+	cout << " is sorted to: ";
+	insertionSort(arr, n);
+	printArray(arr, n);
+	cout << endl;
+
+	return 0;
+}
+
+// This is code is contributed by rathbhupendra

```

Comment:
I am sorry, but I can not accept this PR, since the code is taken from GEEKSFORGEEKS and is not written by you.
Thank you for your contribution. You can try submitting another PR that has your own code. It can be even a single line of code, some documentations, or anything else, but not someone else's code or code taken from a website.


Thanks.